core/vertx/src/main/java/io/thorntail/vertx/DnsClientProducer.java

```diff
@@ -0,0 +1,24 @@
+package io.thorntail.vertx;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
+
+import io.vertx.core.Vertx;
+import io.vertx.core.dns.DnsClient;
+
+/**
+ * Created by bob on 6/22/18.
+ */
+@ApplicationScoped
+public class DnsClientProducer {
+
+    @Produces
+    @ApplicationScoped
```

Comment:
If you produce an `@ApplicationScoped` bean it usually does not make sense to make the producer `@ApplicationScoped` unless it holds some state. In this case, I would use `@Dependent DnsClientProducer` - it will be created to produce a `DnsClient` and destroyed afterwards - and you can even inject `Vertx` directly in the producer method.